test-avocado/lib/create_test_machine.sh

```diff
@@ -0,0 +1,203 @@
+    CTM_IMG=$CTM_NAME.img
+    CTM_MAC=`printf '52:54:00:%02X:%02X:%02X\n' $[RANDOM%256] $[RANDOM%256] $[RANDOM%256]`
+    qemu-img create -f qcow2 $CTM_VMDIRS/$CTM_IMG 20G
+    virt-install --connect=qemu:///system \
+            --network network:default,mac=$CTM_MAC \
+            --initrd-inject=$CTM_ADIR/$CTM_KSF \
+            --extra-args="ks=file:/$CTM_KSF console=tty0 console=ttyS0,115200" \
+            --name=$CTM_NAME \
+            --disk $CTM_VMDIRS/$CTM_IMG \
+            --ram 1028 \
+            --vcpus 1 \
+            --check-cpu \
+            --accelerate \
+            --hvm \
+            --location $CTM_TREE \
+            --noreboot \
+            --nographics
+# 
+
+    is_created $CTM_NAME &>/dev/null
+    setup_vm $CTM_NAME $CTM_PASSWORD
+}
+
+function vm_wait_online(){
+    CTM_MAC=$1
+    CTM_IP=""
+    for foo in `seq 60`;do
+        CTM_IP=`cat /var/lib/libvirt/dnsmasq/default.leases |grep "$CTM_MAC" |head -1 | cut -d ' ' -f 3`
+        echo quit | telnet "$CTM_IP" 22 2>/dev/null | grep -q Connected && return 0
+        echo -n . > /dev/stderr
+        sleep 1
+    done
+    return 1
+}
+
+function is_created(){
+    CTM_NAME=$1
+    if virsh -c qemu:///system list | grep -q $CTM_NAME; then
+        echo "domain $CTM_NAME already exist and running "
+    elif virsh -c qemu:///system list --inactive| grep -q $CTM_NAME; then
+        echo "domain $CTM_NAME already exist and stopped "
+        virsh -c qemu:///system start $CTM_NAME
+        vm_wait_online `vm_get_mac $CTM_NAME`
+    else 
+        echo "domain $CTM_NAME does not exist"
+        return 1
+    fi   
+}
+
+
+function setup_vm(){
+    CTM_HOST="$1"
+    CTM_PASSWD="$2"
+    CTM_IP=`vm_get_ip $CTM_HOST`
+    echo "$CTM_HOST: $CTM_IP"
+    CTM_CTM_LOGIN=$RCTM_USER
+    echo SSHPASS="$CTM_PASSWD" sshpass -e ssh-copy-id $CTM_SSHOPTS $CTM_CTM_LOGIN@$CTM_IP
+    SSHPASS="$CTM_PASSWD" sshpass -e ssh-copy-id $CTM_SSHOPTS $CTM_CTM_LOGIN@$CTM_IP
+    vm_ssh "$CTM_HOST" "$CTM_PREQ"
+}
+
+
+function vm_get_ip(){
+    local CTM_NAME=$1
+    local CTM_MAC=`vm_get_mac $CTM_NAME`
+    is_created $CTM_NAME  &>/dev/null || return 1
+    
+    local CTM_IP=`cat /var/lib/libvirt/dnsmasq/default.leases |grep "$CTM_MAC" |head -1 | cut -d ' ' -f 3`
+    echo $CTM_IP
+}
+function vm_get_mac(){
+    local CTM_NAME=$1
+    local CTM_MAC=`virsh -c qemu:///system dumpxml $CTM_NAME | grep 'mac address' | cut -d\' -f2`
+    echo $CTM_MAC
+}
+
+function vm_get_hostname(){
+    local CTM_NAME=$1
+    local CTM_MAC=`vm_get_mac $CTM_NAME`
+    local CTM_HOSTCTM_NAME=`virsh -c qemu:///system net-dumpxml default | grep $CTM_MAC |sed -r "s/.*name='([^']*).*$/\1/"`
+    echo $CTM_HOSTCTM_NAME
+}
+
+function vm_get_pass(){
+    echo $CTM_PASSWORD
+}
+
+function vm_ssh(){
+    local CTM_HOST=$1
+    shift
+    echo ssh $CTM_SSHOPTS -l $RCTM_USER `vm_get_ip $CTM_HOST` $@
+    ssh $CTM_SSHOPTS -l $RCTM_USER `vm_get_ip $CTM_HOST` $@
+}
+
+
+function virt-create(){
```

Comment:
What about calling this `virt_create`?  With a dash, it looks as if it is a command like `virt-install` or `virt-deploy`.